### DISCUSSION TITLE
YNCA proxy server

### DISCUSSION THREAD

**Discussion by slyoldfox** [mvdwetering/ynca#26] [Ideas]
Just an idea and I haven't read much of the YNCA server.py code yet.

The current implementation of YNCA limits it to 1 device. I was wondering if the `ynca.server` module you have could be turned into a proxy server. The `ynca` spawn the `server` which in turn connects to the device. The `server` would then allow any number of clients to connect to it and multiplex (or broadcast) all commands to the clients.

The idea would be that `yamaha_ynca` uses that proxy server as well, so that you can still connect to the YNCA proxy server to do development or debugging.

**Comment by mvdwetering** [comment]
It should be fairly easy to build something like that, but I don't think it should be part of the server. The server is supposed to emulate a device. It would probably be its own script.

The protocol itself is pretty simple. You would just need to read a line (terminated by `\r\n`) and forward it to the other side of the proxy. I would imagine that there are already solutions out there that do this.

This would probably work pretty well for normal usage, except when multiple Yamaha (YNCA) integrations would be initializing at the same time.

Personally I have always just disabled the integration on my production Home Assistant instance and enabled the one on my Dev instance and reverse it when I am done. Which is not ideal, but because I can use the server for almost all development it is not often needed.

**Reply by slyoldfox** [reply]
Thanks for the reply, keep track of which "client" is requesting what info and only handling that on that specific client would indeed be a bit cumbersome.
